src/olympia/reviewers/views.py

```diff
@@ -1315,6 +1290,36 @@ def allow_resubmission(self, request, **kwargs):
             status_code = status.HTTP_409_CONFLICT
         return Response(status=status_code)
 
+    @action(
+        detail=True,
+        methods=['get'],
+        permission_classes=[AllowAnyKindOfReviewer],
+        url_path=r'file/(?P<file_id>[^/]+)/validation\.json')
+    def json_file_validation(self, request, **kwargs):
+        addon = get_object_or_404(
+            Addon.unfiltered.id_or_slug(kwargs['pk']))
+        file = get_object_or_404(
+            File, version__addon=addon, id=kwargs['file_id'])
+        if file.version.channel == amo.RELEASE_CHANNEL_UNLISTED:
+            if not acl.check_unlisted_addons_reviewer(request):
+                raise PermissionDenied
+        elif not acl.is_reviewer(request, addon):
+            raise PermissionDenied
+        try:
+            result = file.validation
+        except File.validation.RelatedObjectDoesNotExist:
+            raise http.Http404
+        response = JsonResponse({
+            'validation': result.processed_validation,
+            'error': None,
+        })
+        # See: https://github.com/mozilla/addons-server/issues/11048
+        response['Access-Control-Allow-Origin'] = settings.CODE_MANAGER_URL
+        response['Access-Control-Allow-Methods'] = 'GET, OPTIONS'
+        response['Access-Control-Allow-Headers'] = 'Content-Type'
+        response['Access-Control-Allow-Credentials'] = 'true'
```

Comment:
I don't think any of those are necessary any more ? The default for the API should be `Access-Control-Allow-Origin: *`, and with being an API now, it's going to use token-based auth w/ the `Authorization` header, not the session cookie.